File: src/WWS/Order/Web/IntegrationEvents/OrderCreatedEvent.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Web.IntegrationEvents
{
    public class OrderItem
    {
        public long ItemId { get; set; }
        public int Count { get; set; }
    }

    public class OrderCreatedEvent : IOrderCreatedEvent
    {
        public string OrderCode { get; set; }
        public string CustomerId { get; set; }
        public List<OrderItem> Items { get; set; }
    }

    public interface IOrderCreatedEvent
    {
        public string OrderCode { get; set; }
        public string CustomerId { get; set; }
        public List<OrderItem> Items { get; set; }
    }
}

File: src/WWS/Inventory/inventory/src/main/kotlin/hu/bme/aut/inventory/integrationEvent/event/OrderCreatedEvent.kt
package hu.bme.aut.inventory.integrationEvent.event

import com.fasterxml.jackson.annotation.JsonProperty
import java.io.Serializable

data class OrderCreatedEvent(
    @JsonProperty("orderCode") val orderCode: String,
    @JsonProperty("customerId") val customerId: String,
    @JsonProperty("items") val items: List<OrderItem>
) : Serializable

data class OrderItem(
    @JsonProperty("itemId") val itemId: Long,
    @JsonProperty("count") val count: Int,
) : Serializable